### PR TITLE
Migrate to Infinity Cluster

### LINKED ISSUE #1: Migrate to Infinity cluster
Move away from statically hosted site, and instead host on the Infinity ECS cluster!

### PR DESCRIPTION
**Pull request by omkarkaul**
### Issues (if applicable)
Fixes #1 

### Description
Migrates to Infinity cluster by spinning up the following infrastructure:
- ECS Task definition
- ECS Service
- ALB Target group, pointing to the private IP of the Task running on the ECS Service
- ALB Listener rule to point the `www.rakmo.io` and `rakmo.io` domains to the rakmo target group

A multi-stage Dockerfile has been added that builds the React app and serves the built application via an Nginx server

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# dependencies
+/node_modules
+
+# production
+/build
+
+# infrastructure
+infra/*/.terraform*
```
Comment (omkarkaul) [inline .dockerignore:8]: Add new lines

File: .gitignore [hunk 2]
```diff
@@ -23,4 +23,5 @@ yarn-debug.log*
 yarn-error.log*
 
 # infrastructure
+infra/.terraform*
 infra/*/.terraform*
```
Comment (omkarkaul) [inline .gitignore:27]: and here

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+COPY . .
+RUN npm run build
+
+# RUN
+FROM nginx:alpine
+COPY --from=BUILD /app/nginx/nginx.conf /etc/nginx/nginx.conf
+COPY --from=BUILD /app/build /usr/share/nginx/html
+RUN cat /etc/nginx/nginx.conf
+RUN ls -l /usr/share/nginx/html
+EXPOSE 80
```
Comment (omkarkaul) [inline Dockerfile:15]: and here

File: infra/ecr.tf [hunk 4]
```diff
@@ -0,0 +1,3 @@
+resource "aws_ecr_repository" "rakmo_ecr" {
+  name = "rakmo.io"
+}
```
Comment (omkarkaul) [inline infra/ecr.tf:3]: and here

File: infra/elb.tf [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    type = "forward"
+    target_group_arn = aws_lb_target_group.elb_target_group.arn
+  }
+
+  condition {
+    host_header {
+      values = ["rakmo.io", "www.rakmo.io"]
+    }
+  }
+}
```
Comment (omkarkaul) [inline infra/elb.tf:23]: and here

File: infra/security.tf [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        security_groups = [local.elb_security_id]
+    }
+
+    egress {
+        protocol = "-1"
+        from_port = 0
+        to_port = 0
+        cidr_blocks = ["0.0.0.0/0"]
+    }
+}
```
Comment (omkarkaul) [inline infra/security.tf:18]: and here

File: infra/elb.tf [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  priority     = 1
+
+  action {
+    type = "forward"
+    target_group_arn = aws_lb_target_group.elb_target_group.arn
+  }
+
+  condition {
+    host_header {
+      values = ["rakmo.io", "www.rakmo.io"]
```
Comment (omkarkaul) [inline infra/elb.tf:20]: Where is the infra defining these A records pointing to the ALB DNS A record value?

**Comment by NeedsSoySauce** [comment]
did this guy just review his own PR? 🤔

**Pull request closed without merging**